Sources/Valet/SecureEnclaveValet.swift

```diff
@@ -122,38 +122,33 @@ public final class SecureEnclaveValet: NSObject {
     /// - Parameters:
     ///   - object: A Data value to be inserted into the keychain.
     ///   - key: A key that can be used to retrieve the `object` from the keychain.
-    /// - Returns: `false` if the keychain is not accessible.
+    /// - Note: Method will throw a `KeychainError` if an error occurs.
     @objc
-    @discardableResult
-    public func setObject(_ object: Data, forKey key: String) -> Bool {
-        execute(in: lock) {
-            guard let keychainQuery = keychainQuery else {
-                return false
-            }
-            return SecureEnclave.setObject(object, forKey: key, options: keychainQuery)
+    public func setObject(_ object: Data, forKey key: String) throws {
+        try execute(in: lock) {
+            try SecureEnclave.setObject(object, forKey: key, options: try keychainQuery())
         }
     }
 
     /// - Parameters:
     ///   - key: A key used to retrieve the desired object from the keychain.
-    ///   - userPrompt: The prompt displayed to the user in Apple's Face ID, Touch ID, or passcode entry UI.
-    /// - Returns: The data currently stored in the keychain for the provided key. Returns `.itemNotFound` if no object exists in the keychain for the specified key, or if the keychain is inaccessible. Returns `.userCancelled` if the user cancels the user-presence prompt.
-    public func object(forKey key: String, withPrompt userPrompt: String) -> SecureEnclave.Result<Data> {
-        execute(in: lock) {
-            guard let keychainQuery = keychainQuery else {
-                return .itemNotFound
-            }
-            return SecureEnclave.object(forKey: key, withPrompt: userPrompt, options: keychainQuery)
+    ///   -  userPrompt: The prompt displayed to the user in Apple's Face ID, Touch ID, or passcode entry UI.
+    /// - Returns: The data currently stored in the keychain for the provided key.
+    /// - Note: Method will throw a `KeychainError` if an error occurs.
+    @objc
+    public func object(forKey key: String, withPrompt userPrompt: String) throws -> Data {
+        try execute(in: lock) {
+            try SecureEnclave.object(forKey: key, withPrompt: userPrompt, options: try keychainQuery())
         }
     }
 
     /// - Parameter key: The key to look up in the keychain.
-    /// - Returns: `true` if a value has been set for the given key, `false` otherwise. Will return `false` if the keychain is not accessible.
+    /// - Returns: `true` if a value has been set for the given key, `false` otherwise.
     /// - Note: Will never prompt the user for Face ID, Touch ID, or password.
     @objc
     public func containsObject(forKey key: String) -> Bool {
         execute(in: lock) {
-            guard let keychainQuery = keychainQuery else {
+            guard let keychainQuery = try? keychainQuery() else {
```

Comment:
Should this method throw if this throws?

Reply:
I’m not sure! Is `containsObject(forKey` the kind of thing where someone would want/need to understand if an internal error occurred? I guess it’d be good to be able to differentiate between “an error occurred” and “there is no object” though.

Yeah, okay. Let’s make this `throws`. And let’s take the underlying error from `Keychain.containsObject` and throw that too.

(sidebar: I want to get rid of the nullable `keychainQuery` by no longer determining the App ID programmatically. I think enabling App Groups will enable us to remove that & simplify the heck out of this code.)

Reply:
Addressed in 6820428f52986529eca1f9fc42aa49c48847f6f0

Reply:
Resolved in ee56d51a0880476a858e8a71b77f4a37f584f3c4